netket/graph/lattice.py

```diff
@@ -508,14 +454,20 @@ def _get_id_from_dict(
         else:
             raise ValueError("Input needs to be rank 1 or rank 2 array")
 
-    def id_from_position(self, position: PositionT) -> Union[int, Array]:
+    def id_from_position(
+        self, position: PositionT, strict: bool = False
+    ) -> Union[int, Array]:
```

Comment:
I think it is overkill to have a parameter like `strict` to determine whether a function throws an exception or returns an error. Sometimes this is unavoidable for compatibility, but since we are not burdened by that at the moment, I'd rather we make a decision between

1. throwing an exception when a non-existing invalid input is passed;
2. keeping the behavior with `None`.

I'm actually leaning towards to going back to my first version of this method where it was throwing an exception (so, no. 1) on any invalid input.

If in the future we see a use case for looking up a site where the precise position is not known, I feel it'd be a much better API to have something like `closest_site(pos)` and let users decide the distance they are willing to accept, instead of using `id_from_...` which arbitrarily returns `None`.

Reply:
I like 1) more too, because that's the desired behaviour everywhere where it's used now. I'll revert to that soon.